common/src/test/java/gov/cms/ab2d/common/service/PerformanceTestingCoverageOperations.java

```diff
@@ -0,0 +1,275 @@
+     * this is unnecessary.
+     */
+    static {
+        System.setProperty("DB_USERNAME", "ab2d");
+        System.setProperty("DB_PASSWORD", "ab2d");
+        System.setProperty("DB_URL", "jdbc:postgresql://localhost:5432/ab2d");
+    }
+
+    @BeforeEach
+    public void insertContractAndDefaultCoveragePeriod() {
+
+        coverageRepo.deleteAll();
+        coverageSearchEventRepo.deleteAll();
+        coveragePeriodRepo.deleteAll();
+        contractRepo.deleteAll();
+
+        sponsor = dataSetup.createSponsor("Cal Ripken", 200, "Cal Ripken Jr.", 201);
+        contract = dataSetup.setupContract(sponsor, "TST-123");
+
+        period = dataSetup.createCoveragePeriod(contract, 1, 2020);
+    }
+
+    @AfterEach
+    public void cleanUp() {
+        coverageRepo.deleteAll();
+        coverageSearchEventRepo.deleteAll();
+        coveragePeriodRepo.deleteAll();
+        contractRepo.deleteAll();
+
+        if (sponsor != null) {
+            sponsorRepo.delete(sponsor);
+        }
+    }
+
+    @Test
+    void testing() {
+        try {
+            coverageRepo.findCoverageInformation(Collections.singletonList(106), List.of("test-999", "test-1000"));
+        } catch (Exception exception) {
+            exception.printStackTrace();
+        }
+        System.out.println("woohoo");
+    }
+
+    /**
+     * Performance test for inserts, jack this number up really high
+     */
+    @Disabled
+    @DisplayName("Performance testing for inserts")
+    @Test
+    void insertPerformanceUsingConnection() {
+
+        CoverageSearchEvent inProgress = new CoverageSearchEvent();
+        inProgress.setOldStatus(JobStatus.SUBMITTED);
+        inProgress.setNewStatus(JobStatus.IN_PROGRESS);
+        inProgress.setDescription("testing");
+        inProgress.setCoveragePeriod(period);
+
+        coverageSearchEventRepo.saveAndFlush(inProgress);
+
+        // Raise datapoints to stress database
+        List<Long> timings = insertData(inProgress, 1_000_000, 10, true);
+
+        long averageTime = timings.stream().reduce(0L, Long::sum) / timings.size();
+        System.out.println("Average milliseconds " + averageTime);
+        System.out.println("Times " + timings.stream().map(Object::toString).collect(joining(", ")));
+    }
+
+    /**
+     * Performance test for inserts, jack this number up really high
+     */
+    @Disabled
+    @DisplayName("Performance testing for reads")
+    @Test
+    void readPerformanceWithJPAPaging() {
+
+        loadDBWithFakeData(2_000_000);
+
+        int queries = 100;
+        int threads = 16;
+        int pageSize = 10000;
+        int pages = 2_000_000 / 10000;
+
+        ExecutorService executor = Executors.newFixedThreadPool(threads);
+
+        Random random = new Random();
+
+        List<Long> timing = new ArrayList<>(queries * threads);
+
+        Runnable select = () -> {
+            int i = 0;
+
+            while (i < queries) {
```

Comment:
Style-wise, this is a for loop.  Please use **idx** (or something slightly more verbose than **i**).. Searching for **i** in a code base is a real pain.  Searching for **idx** is much easier.

Reply:
Using i seems fine to me it's pretty universally used as a index all over the JDK, don't think we need to search for it, plus it's shorter to write